### PR TITLE
enable boot diagnostics

### PR DESCRIPTION
**Pull request by mjudeikis**
Enables boot diagnostics.
Add storage account for it.

### CODE REVIEW

**Comment by jim-minter** [comment]
Why do you want to do this?

**Comment by jim-minter** [comment]
In general I want to enable geneva logging on the RP VMs and ensure it's working properly as the logging strategy.

File: deploy/rp-production-nsg.json [hunk 1]
```diff
@@ -33,6 +33,9 @@
         }
     ],
     "outputs": {
-        "rpServicePrincipalId": "[reference(resourceId('Microsoft.ManagedIdentity/userAssignedIdentities', 'rp-identity'), '2018-11-30').principalId]"
+        "rpServicePrincipalId": {
+            "type": "string",
```
Comment (jim-minter) [inline deploy/rp-production-nsg.json:37]: I've picked this part of the PR up separately - thanks for spotting it.

**Comment by mjudeikis** [comment]
> Why do you want to do this?

First, when facing issues this is one of the first thing VMSS teams are asking for. 
Second for our own debugging when VMSS are stuck or nor responding it helps to debug.

Overall it is to cover future issues we will face.

**Comment by jim-minter** [comment]
keep the branch if you like, but I don't believe we need this now.

**Pull request closed without merging**